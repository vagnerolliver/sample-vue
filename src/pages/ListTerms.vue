<template>
  <div>
    <table class="table">
      <thead>
      <tr>
        <th class="is-uppercase">{{ title }}</th>
        <th class="is-uppercase">{{ description }}</th>
      </tr>
      </thead>
      <tbody>
        <tr v-for="(term, index) in terms">
          <th>{{ term.name }}</th>
          <td>{{ term.value }}</td>
        </tr>
      </tbody>
    </table>
  </div>
</template>

<script>
  import { getTerms } from "@/untils/crud";

  export default {
    name: 'ListTerms',
    data () {
      return {
        title: '',
        description: '',
        terms: []
      }
    },
    created () {
      this.title = 'Título da minha pagina!'
      this.description = 'Minha descrição'
    },
    mounted () {
      this.listTerms();
    },
    methods: {
      listTerms: function() {
        return getTerms('terms')
          .then(response => {
            this.terms = response
          })
      }
    }
  }
</script>


<style lang="sass">

  $color: red

  h1
    color: blue

</style>
