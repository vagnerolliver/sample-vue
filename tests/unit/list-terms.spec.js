import ListTerms from '@/pages/ListTerms.vue'



describe('List Terms', () => {

  describe('should find by Terms', () => {

    // Inspect the raw component options
    test('has a created hook', () => {
      expect(typeof ListTerms.created).toBe('function')
    })
  });
})
