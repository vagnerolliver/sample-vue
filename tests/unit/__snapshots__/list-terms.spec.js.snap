// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot renders correctly 1`] = `
<div>
  <table
    class="table"
  >
    <thead>
      <tr>
        <th
          class="is-uppercase"
        >
          Título da minha pagina!
        </th>
         
        <th
          class="is-uppercase"
        >
          Minha descrição
        </th>
      </tr>
    </thead>
     
    <tbody>
      <tr>
        <th>
          BOTAO_CANCELAR
        </th>
         
        <td>
          Cancelar
        </td>
      </tr>
       
      <tr>
        <th>
          BOTAO_DELETAR
        </th>
         
        <td>
          Deletar
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
